lib/src/linking.rs

```diff
@@ -6,38 +6,83 @@ use {
         wiggle_abi, Error,
     },
     std::collections::HashSet,
-    wasmtime::{GuestProfiler, Linker, Store, UpdateDeadline},
+    wasmtime::{GuestProfiler, Linker, Store, StoreLimits, StoreLimitsBuilder, UpdateDeadline},
     wasmtime_wasi::{preview1::WasiP1Ctx, WasiCtxBuilder},
     wasmtime_wasi_nn::WasiNnCtx,
 };
 
-#[derive(Default)]
 pub struct Limiter {
     /// Total memory allocated so far.
     pub memory_allocated: usize,
+    /// The internal limiter we use to actually answer calls
+    internal: StoreLimits,
+}
+
+impl Default for Limiter {
+    fn default() -> Self {
+        Limiter::new(1, 1)
+    }
+}
+
+impl Limiter {
+    fn new(max_instances: usize, max_tables: usize) -> Self {
+        Limiter {
+            memory_allocated: 0,
+            internal: StoreLimitsBuilder::new()
+                .instances(max_instances)
+                .memories(1)
+                .memory_size(128 * 1024 * 1024)
+                .table_elements(98765)
+                .tables(max_tables)
+                .build(),
+        }
+    }
 }
 
 impl wasmtime::ResourceLimiter for Limiter {
     fn memory_growing(
         &mut self,
         current: usize,
         desired: usize,
-        _maximum: Option<usize>,
+        maximum: Option<usize>,
     ) -> anyhow::Result<bool> {
         // Track the diff in memory allocated over time. As each instance will start with 0 and
         // gradually resize, this will track the total allocations throughout the lifetime of the
         // instance.
         self.memory_allocated += desired - current;
-        Ok(true)
+        // limit the amount of memory that an instance can use to (roughly) 128MB, erring on
+        // the side of letting things run that might get killed on Compute, because we are not
+        // tracking some runtime factors in this count.
+        self.internal.memory_growing(current, desired, maximum)
```

Comment:
What do you think about incrementing `memory_allocated` only after a successful call to `self.internal.memory_growing(..)`? That way we would only report the memory that was successfully allocated.